packages/web-components/fast-foundation/src/menu-item/menu-item.template.ts

```diff
@@ -30,25 +42,21 @@ export function menuItemTemplate(
             ${when(
                 x => x.role === MenuItemRole.menuitemcheckbox,
                 html<FASTMenuItem>`
-                    <div part="input-container" class="input-container">
-                        <span part="checkbox" class="checkbox">
-                            <slot name="checkbox-indicator">
-                                ${options.checkboxIndicator || ""}
-                            </slot>
-                        </span>
-                    </div>
+                    <span part="checkbox-indicator" class="checkbox-indicator">
+                        <slot name="checkbox-indicator">
+                            ${options.checkboxIndicator || checkboxIndicator}
+                        </slot>
+                    </span>
                 `
             )}
             ${when(
                 x => x.role === MenuItemRole.menuitemradio,
                 html<FASTMenuItem>`
-                    <div part="input-container" class="input-container">
-                        <span part="radio" class="radio">
-                            <slot name="radio-indicator">
-                                ${options.radioIndicator || ""}
-                            </slot>
-                        </span>
-                    </div>
+                    <span part="radio-indicator" class="radio-indicator">
+                        <slot name="radio-indicator">
+                            ${options.radioIndicator || radioIndicator}
+                        </slot>
+                    </span>
```

Comment:
These are both typically in the same spot because an item is *either* a `menuitem`, `menuitemcheckbox`, or `menuitemradio`, so I'm wondering if we can simplify this with something that serves a single indicator but sets the class and part based on its role. So a single conditional check with the part/class set based on the role. Thoughts? I think it simplifies the template and still accomplishes the same thing.

Reply:
Actually, I think we may only need *one* of these and I think it should likely still be called `checked-indicator`. 

While I know that the initial `fast-components` made these look like a checkbox and an radio, the actual indicator represents the `checked` state of either an element with one of the checkable roles. I'd like to dig in here, because I think this can simplify and while someone may choose to make it look like one or the other, that's also primarily state based that can be handled in styles or using a when directive from the view to manage sending alternative svg's here. In other implementations, it may be the same SVG, but the behavior only allows single selection.

Reply:
See my comment on `progress` for one potential way to slim down this block by removing the `slot`. I don't see a case for individual components to slot in a custom indicator here.

I do think it's beneficial to be able to provide the icons individually, the same icon if that's what your design called for, but I think it's good UX to have different icons. I'd hate to see the css get messier to accommodate a minor reduction.

Reply:
Again, I don't think that is prohibited by having a singular slot. These could be handled in configuration using a when directive without needing to hoist that requirement to *all* implementations. If the goal is whether we can accomplish what you've stated above I think we should use the most concise method of addressing that. I commented below but the removal of slots in general is a non-starter. The options don't replace it, they provide a method of configuring a global default. That helps for "menu" but if I need to configure something else there in a one-off scenario, the slot is still massively more useful.

Reply:
Yes ^ - My big question here is whether the better ergonomic is baking the conditional rendering in or letting that be handled by the person implementing. On one hand, in the case I don't need two I need to pass the same thing twice. Alternatively, if I have two different SVG's I can handle my own when, right? Does one match our composition value more than the other?

Reply:
I'd love to find a way to slim down the template here if that's what @EisenbergEffect's last comment suggests. I'm not sure I am seeing the exact change though.

From a component framework perspective, the person authoring the component doesn't know if someone is going to use checkbox or radio or not. Since the component supports this to fully support the ARIA spec it should also have a clear way to have two different indicators. This should be easy without having to worry about it for each menu you implement.

Reply:
I've consolidated this to a single conditional using the pattern from `Flipper`.

I understand that someone may not always need checked functionality, and the only feasible solution I see for that is using the template builder to exclude it. For now we have that role capability built in.

From a visual perspective, it's bad usability for two things that operate differently to have the same look, so I can't in good conscience agree with a solution where the easy default is to treat them the same. The fact that they both relate to the `checked` state is not the sole decision or the ARIA spec could have omitted the two roles and encouraged people to implement `radio` functionality in code.

This is at least better than what we had, and if we want to consider changing this, though my vote is not to, we should pull it into a new issue.